modules/ROOT/pages/apikit-workflow-convert-existing.adoc

```diff
@@ -1,104 +1,239 @@
-
-Now, you are ready to run the new project and simulate calls using API Console.
+The *New Mapping* dialog appears.
+. Use the drop-down to map the resources to actions.
++
+* In the *Resource* drop-down, select `/users/userbyid`.
++
+* In the *Action* drop-down, select `Get`.
++
+* In the *Flow* drop-down, select the flow that contains the get users action: `GET-User-By-Id-FLOW`.
++
+* Click *OK*.
+
+[source,xml,linenums]
+----
+<mule>
+    <apikit:config outboundHeadersMapName="outboundHeadersMapName" httpStatusVarName="httpStatus" name="Router" api="api/api.raml" >
+		<apikit:flow-mappings >
+			<apikit:flow-mapping resource="/users" action="get" flow-ref="GET-Users-FLOW" />
+			<apikit:flow-mapping resource="/users/userbyid" action="get" flow-ref="GET-User-By-Id-FLOW" />
+		</apikit:flow-mappings>
+	</apikit:config>
+    <http:listener-config name="HTTP_Listener_config">
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<db:config name="Database_Config">
+		<db:my-sql-connection host="localhost" port="3306" user="admin" password="admin" database="usersdb" />
+	</db:config>
+    <flow name="apiFlow">
+		<http:listener config-ref="HTTP_Listener_config" path="/api/*"/>
+		<apikit:router config-ref="Router"/>
+	</flow>
+	<flow name="GET-Users-FLOW">
+		<db:select config-ref="Database_Config">
+			<db:sql ><![CDATA[SELECT * FROM USERS]]></db:sql>
+		</db:select>
+	</flow>
+	<flow name="GET-User-By-Id-FLOW">
+		<db:select config-ref="Database_Config" >
+			<db:sql ><![CDATA[SELECT * FROM USERS WHERE :userIdParameter == userID]]></db:sql>
+			<db:input-parameters ><![CDATA[#[{userIdParameter: attributes.queryParams.id}]]]></db:input-parameters>
+		</db:select>
+	</flow>
+</mule>
+----
+
+== Scaffolding Approach
+
+Follow the process at xref:apikit-workflow-convert-existing.adoc#step-1-design-your-api-specification[]. When completed, include an alternative workflow for scaffolding your API specification.
+
+=== Step 2: Scaffold Your API Specification
+
+Scaffold your API to auto-generate the main, console, and resources flows:
+
+. In *Package Explorer*, right-click `/src/main/resources/api/api.raml`.
+. Select *Mule*.
+. Click *Generate Flows from Local REST API*.
+
+A new XML file with the same name as the API is created containing the scaffolded flows.
+
+NOTE: With this approach you get the APIkit router configuration, the error handler, and the console flow out of the box. You also get a flow that maps each resource.
+
+=== Step 3: Reference Flows
+
+At this point, you have some options:
+
+* Delete the scaffolded flows for each resource and map the original flows as mentioned in xref:apikit-workflow-convert-existing.adoc#step-3:-map-api-resources-actions-to-flows
+[].
+* Remove any operations from the scaffolded flows and add a flow reference instead. By choosing this option you introduce an additional layer of indirection, which can be inconvenient in some scenarios.
```

Comment:
any operations > all operations? any operation?

What do you mean by an additional layer of indirection?

Reply:
Requesting assistance to SME.

Reply:
It's all operations. Making the necessary changes. 
The "layer of indirection" means that you put all operations in one flow, and you reference the other flows to the one that has the operations. I'll try to convey such meaning in a clear way.